src/orion/core/io/database/ephemeraldb.py

```diff
@@ -0,0 +1,440 @@
+        for key, value in query.items():
+            if not self.match_key(key, value):
+                return False
+
+        return True
+
+    def match_key(self, key, value):
+        """Test if a data corresponding to the given key is in agreement with the given
+        value based on the operator defined within the key.
+
+        Default operator is equal when no operator is defined.
+        Other operators could be $in, $gte, $gt. They are defined
+        in the last section of the key. For example: `abc.def.$in` or `abc.def.$gte`.
+        """
+        if key.split(".")[-1] in self.operators:
+            operator = key.split(".")[-1]
+            key = ".".join(key.split(".")[:-1])
+
+            return key in self and self.operators[operator](self[key], value)
+
+        return key in self and self[key] == value
+
+    def _validate_keys(self, keys):
+        """Verify that all keys are 0 or 1 (with exception of _id) and convert them.
+
+        For simplicity, when keys are 0, the inverse set of keys for 1s is computed.
+
+        .. note ::
+
+            _id is set to 1 if not specified. Only _id may be set to 0 if other keys are set to 1.
+        """
+        keys_without_id = [key for key in keys if key != '_id']
+        n_keys = sum(keys[key] for key in keys_without_id)
+        if n_keys != 0 and n_keys != len(keys_without_id):
+            raise ValueError(
+                'Cannot mix selection with 1 and 0s except for _id: {}'.format(keys))
+
+        if n_keys == 0:
+            new_keys = dict((key, 1) for key in self._data.keys() if key not in keys)
+            new_keys['_id'] = keys.get('_id', 1)
+            keys = new_keys
+
+        return keys
+
+    def select(self, keys):
+        """Only select or only drop the specified keys
+
+        For a pair (key, value) in the dictionnary, value=0 means the key will not be included
+        while value=1 means it will.
+
+        All specified keys should be 0 or 1. They cannot have different values with the exception
+        of _id which can be specified to 0 while the others are at 1.
+
+        Parameters
+        ----------
+        keys: dict
+            Pairs of keys and 0 or 1s. When a key is associated with 1, it is kept in the selection,
+            otherwise it is dropped.
+
+        """
+        if not keys:
+            return _unflatten(self._data)
+
+        keys = _flatten(keys)
+        keys = self._validate_keys(keys)
+
+        selection = dict()
+
+        def key_is_match(key, selected_key):
+            """Test if key matches the selected key
+
+            key_is_match(abc.def.ghi, abc.def.ghi) -> True
+            key_is_match(abc.def.ghi, abc.def) -> True
+            key_is_match(abc.def.ghi, abc.de) -> False
+            key_is_match(abc.def.ghi, xyz) -> False
+            """
+            if not key.startswith(selected_key):
+                return False
+
+            if not key.replace(selected_key, ''):
```

Comment:
This is saying `if key == selected_key`, right?

Reply:
OMG. I guess it was getting late :laughing: